src/main/java/org/dependencytrack/persistence/ProjectQueryManager.java

```diff
@@ -473,18 +502,21 @@ public Project updateProject(Project transientProject, boolean commitIndex) {
         project.setPurl(transientProject.getPurl());
         project.setSwidTagId(transientProject.getSwidTagId());
 
-        if (project.isActive() && hasActiveChild(project) && (transientProject.isActive() == null || !transientProject.isActive()) ){
-            LOGGER.warn(project + " cannot be set to inactive, if active children are present. " + project + " remains active");
+        if (project.isActive() && !Boolean.TRUE.equals(transientProject.isActive()) && hasActiveChild(project)){
+            throw new IllegalArgumentException("Project cannot be set to inactive if active children are present.");
         } else {
             project.setActive(transientProject.isActive());
         }
```

Comment:
```suggestion
        }
        project.setActive(transientProject.isActive());
```
Kannst dir das else sparen wenn im if eine exception fliegt